### PR TITLE
[Cleanup] Move opcode data initialization from headers into dedicated C file

### PR DESCRIPTION
**Pull request by happy-barney**
Planned follow-ups (long run, this PR has been waiting almost 2 years):

1. -use designated array initializers (should not make problems in internal C-files)-
2.  extend data with "not supported" message, moving for example usage of `PL_no_dir_func` from `pp_sys.c` into single `PL_not_supported` function 
3. provide accessor functions instead of direct data access
  - allows introduction of custom op codes, may be interesting for some grammar extension
4. merge data into single structure
  - allows unified handling of unavailable op, see for example difference between `read` (handled in `opcode.h`) and `readdir` / `open_dir` (handled in `pp_sys.c`)

Example of structure declaration (here using designated initializers for readability)
```
#define PL_no_dir_func  "Unsupported directory function \"%s\" called"

#if defined(Direntry_t) && defined(HAS_READDIR)
#  define PL_PLATFORM_PP_READDIR Perl_pp_readdir
#else
#  define PL_PLATFORM_PP_READDIR Perl_pp_unsupported
#endif

  [OP_READDIR] = {
    .op_name = "readdir",
    .op_desc  = "readdir",
    .op_ppaddr = PL_PLATFORM_PP_READDIR,
    .op_check = Perl_ck_fun,
    .op_opargs = 0x00006b00,
    .op_private_labels = 'U',
    .op_private_bitdef_ix = 0,
    .op_unsupported = PL_no_dir_func,
  }
```

### CODE REVIEW

**Comment by jkeenan** [comment]
Can you address the porting test failures reported in https://github.com/Perl/perl5/actions/runs/10548400876/job/29222271299?pr=22540, then re-push?

Issuing a pull request that is going to fail multiple tests (porting or otherwise) is not helpful to potential code reviewers.

**Comment by happy-barney** [comment]
> Can you address the porting test failures reported in https://github.com/Perl/perl5/actions/runs/10548400876/job/29222271299?pr=22540, then re-push?
> 

just re-pushed ... had to rethink my  approach due tests relying in symbols defined in `globals.o`

**Comment by tonycoz** [comment]
> 1. use designated array initializers (should not make problems in internal C-files)

We have developers who build perl with C++ compilers, and designated array initializers aren't in any C++ standard (g++ allows it anyway, MSVC doesn't).

File: regen/opcode.pl [hunk 1]
```diff
@@ -821,27 +830,12 @@ sub print_PL_op_private_tables {
...
-
 /* PL_op_private_labels[]: the short descriptions of private flags.
  * All labels are concatenated into a single char array
  * (separated by \\0's) for compactness.
  */
 
-EXTCONST char PL_op_private_labels[] = {
+const char * PL_op_private_labels = (const char []) {
 $PL_op_private_labels
 };
```
Comment (tonycoz) [inline regen/opcode.pl:840]: Compound literal syntax is disallowed by perlhacktips for C++ compatibility.

This also changes PL_op_private_labels from const to non-const.

(it was a const array of char, this changes it to a non-const pointer to const char.)

Similarly below.

I don't understand why you made this change.

Comment (happy-barney) [reply regen/opcode.pl:840]: It was part of discussion about my earlier commit (2 years ago) - to move variable initialization from `.h` files to `.c` files, get rid of `INIT` macro, and redefinition of various macros for `globals.c`

**Comment by happy-barney** [comment]
> > 1. use designated array initializers (should not make problems in internal C-files)
> 
> We have developers who build perl with C++ compilers, and designated array initializers aren't in any C++ standard (g++ allows it anyway, MSVC doesn't).

shame, such a nice feature improving readability of code :-(
updating PR comment

**Pull request closed without merging**